src/core/public/integrations/moment/moment_service.ts

```diff
@@ -35,7 +36,10 @@ export class MomentService implements CoreService {
   public async setup() {}
 
   public async start({ uiSettings }: StartDeps) {
-    const setDefaultTimezone = (tz: string) => moment.tz.setDefault(tz);
+    const setDefaultTimezone = (tz: string) => {
+      const zone: string | undefined = get(moment.tz.zone(tz), 'name');
+      moment.tz.setDefault(zone);
```

Comment:
could add a test that `moment.tz.setDefault(undefined);` doesn't remove previously set timezone?